APIv2/upload_example.md

```diff
@@ -46,7 +46,7 @@ file_id = json.loads(resp.content)['location'].rsplit('/', 1)[1]
 endpoint = 'account/articles/{}/files/{}'.format(article_id, file_id)
 resp = requests.get(BASE_URL.format(endpoint=endpoint), headers=HEADERS)
 
-url = '{upload_url}'.format(**json.loads(resp.content))
+url = '{upload_url}'.format(**json.loads(resp.content['upload_url']))
```

Comment:
Well `resp.content` contains a **JSON** formatted string. After the `json.loads` call it turns into a dictionary that contains as one of its first level keys _upload_url_. Using the magic operator `**` it is then unpacked as parameters to the `format` function. So in equivalence the line you changed would be similar to this:
```
data = json.loads(resp.content)
upload_url = data['upload_url']
url = '{upload_url}'.format(upload_url=upload_url) 
```

The run code would probably just fail at this point with an error stating that you can't index with something different than a number; which may be rather cryptic.

The cause for this weirdness may be to the fact that probably the _url_ was composed out of several parts that were all found in `resp.content` and that may have made sense for example if `resp.content` was exactly 
```
'{"upload_url": "some_value", "upload_token": "some_token"}'
```
and we'd have to let the `url` be made of **upload_url**/**upload_token**, then that line would have been: 
```
url = '{upload_url}/{upload_token}'.format(**json.loads(resp.content))
```
and everything would have worked perfectly, would have made sense and would have been a oneliner.

My opinion is that this line **would probably just be better written as**:
```
url = str(json.loads(resp.content)['upload_url'])
```